GoogleForCreators/web-stories-wp
# Pre-publish Checklist: Add toggle to enable auto media optimization

## Context

A toggle is wanted in the pre-publish checklist that allows users to allow for auto video optimization. Only the toggle and text was done in this pr.

[Designs](https://www.figma.com/file/1YM3jA9h4QGc4xLdyaXIJS/Stories-Sprint-1.7?node-id=7203%3A23038)

<img src="https://user-images.githubusercontent.com/22185279/115473164-71e92c00-a1f8-11eb-94b7-e71fcf007673.png" width="350px" />

## Summary

- Add toggle to checklist
- Wire it up to api so that users can clicky clicky
- Add unit + karma tests

## Relevant Technical Choices

- made some small styling updates to the toggle for better usability.
- using the same flag as in #7130

## To-do


## User-facing changes

|Before|After|
|--|--|
|![image](https://user-images.githubusercontent.com/22185279/115473375-dc9a6780-a1f8-11eb-9c99-ced78acd5116.png)|![image](https://user-images.githubusercontent.com/22185279/115473353-cdb3b500-a1f8-11eb-8e1d-67f7dbfef572.png)|

|Accessibility|RTL|
|--|--|
|<img src="https://user-images.githubusercontent.com/22185279/115473703-6b0ee900-a1f9-11eb-9837-bd3cba4fc7e3.png" />|![image](https://user-images.githubusercontent.com/22185279/115474013-c8a33580-a1f9-11eb-81a5-05b8f6dcf2cf.png)|


## Testing Instructions

1. Make sure the `Video optimization pre-publish checklist` flag is turned on (under experiments)
2. Make sure that `Video Optimization` is not checked in the dashboard settings
3. Go to editor
4. Open PPC
5. Should see toggle. Clicking the `Settings` link should take you to the dashboard settings.
6. Click toggle
7. Reload page
8. Toggle should be gone
9. Go to dashboard settings - should see the `Video Optimization` boxed checked 🥳  

### QA






This PR can be tested by following these steps:

1. Make sure the `Video optimization pre-publish checklist` flag is turned on (under experiments)
2. Make sure that `Video Optimization` is not checked in the dashboard settings
3. Go to editor
4. Open PPC
5. Should see toggle. Clicking the `Settings` link should take you to the dashboard settings.
6. Click toggle
7. Reload page
8. Toggle should be gone
9. Go to dashboard settings - should see the `Video Optimization` boxed checked 🥳  

### UAT






## Reviews

### Does this PR have a security-related impact?

no

### Does this PR change what data or activity we track or use?

Yes - this adds a tracker when clicking the `Settings` link in the toggle label.

### Does this PR have a legal-related impact?

No

## Checklist




---



Fixes #7256
